src/core/Slider/Slider.tsx

```diff
@@ -0,0 +1,454 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import cx from 'classnames';
+import React from 'react';
+import { useTheme } from '../utils/hooks/useTheme';
+import '@itwin/itwinui-css/css/slider.css';
+import { TooltipProps } from '../Tooltip';
+import { CommonProps } from '../utils/props';
+import { getBoundedValue } from '../utils/common';
+import { Track } from './Track';
+import { Thumb } from './Thumb';
+
+/**
+ * Determines which segments are shown with color.
+ * 'auto' - segment display is based on number of values.
+ * 'none' - no colored tracks are displayed.
+ * 'odd-segments'- colored tracks shown in segments 1,3,5, etc. Default if number of Thumbs values are odd.
+ * 'even-segments'- colored tracks shown in segments 2,4,6, etc. Default if number of Thumbs values are even.
+ */
+export type TrackDisplayMode =
+  | 'auto'
+  | 'none'
+  | 'odd-segments'
+  | 'even-segments';
+
+/** Private Utility Functions */
+const getPercentageOfRectangle = (rect: DOMRect, pointer: number) => {
+  const position = getBoundedValue(pointer, rect.left, rect.right);
+  return (position - rect.left) / rect.width;
+};
+
+const getClosestValueIndex = (values: number[], pointerValue: number) => {
+  if (1 === values.length) {
+    return 0;
+  }
+  const distances = values.map((value) => Math.abs(value - pointerValue));
+  const smallest = Math.min(...distances);
+  return distances.indexOf(smallest);
+};
+
+const getDefaultTrackDisplay = (
+  trackDisplayMode?: TrackDisplayMode,
+  values?: number[],
+) => {
+  const numValues = values?.length ?? 0;
+  if (undefined !== trackDisplayMode && 'auto' !== trackDisplayMode) {
+    return trackDisplayMode;
+  }
+
+  return numValues % 2 ? 'even-segments' : 'odd-segments';
+};
+
+const roundValueToClosestStep = (value: number, step: number, min: number) => {
+  return Math.round((value - min) / step) * step + min;
+};
+
+const formatNumberValue = (
+  value: number,
+  step: number,
+  numDecimals: number,
+) => {
+  if (Number.isInteger(step)) {
+    return value.toFixed(0);
+  }
+  return value.toFixed(numDecimals);
+};
+
+/**
+ * Properties for Slider component
+ */
+export type SliderProps = {
+  /**
+   * Set focus on first thumb in slider element.
+   * @default false
+   */
+  setFocus?: boolean;
+  /**
+   * Minimum slider value.
+   * @default 0
+   */
+  min?: number;
+  /**
+   * Maximum slider value.
+   * @default 100
+   */
+  max?: number;
+  /**
+   * Array of one or more values to show.
+   */
+  values: number[];
+  /**
+   *  Determines which segments are shown with color.
+   * 'none' - no colored tracks are displayed.
+   * 'auto' - no colored tracks are displayed.
```

Comment:
This comment is a duplicate.. I would suggest removing it from types and leaving it here maybe?